framework/io/obj_io.cc

```diff
@@ -0,0 +1,392 @@
+// SPDX-FileCopyrightText: 2024 The OpenSn Authors <https://open-sn.github.io/opensn/>
+// SPDX-License-Identifier: MIT
+
+#include "framework/io/obj_io.h"
+#include "framework/runtime.h"
+#include "framework/logging/log.h"
+#include <fstream>
+#include <algorithm>
+
+namespace opensn
+{
+
+std::shared_ptr<UnpartitionedMesh>
+ObjIO::FromFile(const UnpartitionedMesh::Options& options)
+{
+  const std::string fname = "ObjIO::ReadFromWavefrontOBJ";
+
+  // Opening the file
+  std::ifstream file;
+  file.open(options.file_name);
+  if (not file.is_open())
+  {
+    log.LogAllError() << "Failed to open file: " << options.file_name << " in call "
```

Comment:
Show we `throw` here? I think this is the only place I see where`Exit` is called instead of a `throw`.

Reply:
This way handling the error of "unable to open file" seems to be used in more places in the whole code. I will patch it with `throw` where I think it is appropriate - like these IO routines. The VTK reader does throw, but OBJ and Gmsh `Exit()`.